### DISCUSSION TITLE
Gallery：可以切换 4+ 主题风格

### DISCUSSION THREAD

**Discussion by visiky** [antvis/G2#4397] [Ideas]
### 需求

增加一个简单的 dashboard 页面，可以切换查看不同的主题风格：包括 light、dark、pattern、rough 以及其它

### 方案

增加一个页面，提供 4+ 主题风格，类似于 [highcharts](https://www.highcharts.com/demo/brand-dark) 的形式，暂时不提供精细化的主题构建器
- 技术层面上是直接提供风格选择的 swtich 组件，代码上切换 theme type
- Demos 可以从 examples 中选取（直接把代码复制一份），方便关联跳转到 examples 的案例进行编码

#### AnyCharts
<img width="600" alt="image" src="https://user-images.githubusercontent.com/15646325/204730527-2251ce96-842e-430d-88dc-87d9a570c955.png">
<img width="600" alt="image" src="https://user-images.githubusercontent.com/15646325/204730633-fb936fc1-1208-46a0-84b5-bb8a095d15dd.png">

#### Nivo、ECharts

nivo、echarts 和之前 AntV 的主题构建器还挺像的，比较精细化

<img width="600" alt="image" src="https://user-images.githubusercontent.com/15646325/204731052-cfb6b19e-bfa7-4dd5-9924-a652915489e6.png">

#### HighCharts

<img width="600" alt="image" src="https://user-images.githubusercontent.com/15646325/204731555-3844efd2-95ad-4e53-a48f-b06ee4310c67.png">

**Comment by visiky** [comment]
预览效果，后续还可以做更多事情

<img width="1435" alt="image" src="https://user-images.githubusercontent.com/15646325/205235768-bba8c3b1-ab36-415b-accc-4c6e9c0f4b17.png">

<img width="1438" alt="image" src="https://user-images.githubusercontent.com/15646325/205290470-22b964fe-6725-4e5f-8b32-f6a98bff594e.png">

<img width="1429" alt="image" src="https://user-images.githubusercontent.com/15646325/205318924-997b45a2-208a-469c-a104-99c684c6c181.png">

**Reply by visiky** [reply]
`source` badge 有点丑，考虑移除掉

**Comment by visiky** [comment]
1、增加「拷贝」
2、不重新渲染，通过 update 更新主题
3、细节问题
4、黑色背景
5、学术风格作为 G2 官网提供的

**Reply by hustcc** [reply]
最终还是做成主题构建器。

**Reply by visiky** [reply]
参考：https://theme-set.antv.vision/
